### PR TITLE
fix: keep unhandledRejectionError err object stack

### PR DESCRIPTION
**Pull request by fengmk2**
##### Checklist

- [x] `npm test` passes
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message follows commit guidelines

### CODE REVIEW

**Comment by fengmk2** [comment]
err 可能是一个 object，但是它带了 stack

**Comment by shaoshuai0102** [comment]
什么情况下会有这种 object

File: lib/egg.js [hunk 1]
```diff
@@ -219,7 +219,12 @@ class EggApplication extends EggCore {
 
   _unhandledRejectionHandler(err) {
     if (!(err instanceof Error)) {
+      // err maybe an object and contain stack property, just copy the stack to the new error instance
```
Comment (dead-horse) [inline lib/egg.js]: 严格一点判断吧， message, stack, name 三个属性。

File: lib/egg.js [hunk 2]
```diff
@@ -219,7 +219,14 @@ class EggApplication extends EggCore {
 
   _unhandledRejectionHandler(err) {
     if (!(err instanceof Error)) {
-      err = new Error(String(err));
+      const newError = new Error(String(err));
+      // err maybe an object, try to copy the name, message and stack to the new error instance
+      if (err) {
+        if (err.name) newError.name = err.name;
+        if (err.message) newError.message = err.message;
+        if (err.stack) newError.stack = err.stack;
```
Comment (fengmk2) [inline lib/egg.js]: @dead-horse 尽量还原这3个属性

**Comment by fengmk2** [comment]
Also fixed in koa onerror https://github.com/koajs/onerror/pull/20

**Pull request merged**